applications/crossbar/src/modules/cb_search.erl

```diff
@@ -125,6 +164,86 @@ validate_search(Context, Type, Query) ->
             Context1
     end.
 
+validate_search(Context, _Type, _Query, 'undefined') ->
+    cb_context:add_validation_error(
+      <<"v">>
+      ,<<"required">>
+      ,wh_json:from_list([{<<"message">>, <<"Search needs a value to search with">>}])
+      ,Context
+     );
+validate_search(Context, Type, Query, <<_/binary>> = Value) ->
+    search(Context, Type, Query, Value);
+validate_search(Context, Type, Query, Value) ->
+    case wh_util:is_true(Value) of
+        'true' -> validate_search(Context, Type, Query, <<>>);
+        'false' -> validate_search(Context, Type, Query, 'undefined')
+    end.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec validate_multi(cb_context:context(), api_binary()) -> cb_context:context().
+-spec validate_multi(cb_context:context(), ne_binary(), wh_proplist()) -> cb_context:context().
+validate_multi(Context, 'undefined') ->
+    cb_context:add_validation_error(
+        <<"t">>
+        ,<<"required">>
+        ,wh_json:from_list([{<<"message">>, <<"Search needs a document type to search on">>}])
+        ,Context
+    );
+validate_multi(Context, Type) ->
+    QS = cb_context:query_string(Context),
+    case wh_json:to_proplist(wh_json:delete_key(<<"t">>, QS)) of
+        [_|_]=Props -> validate_multi(Context, Type, Props);
+        _Other ->
+            cb_context:add_validation_error(
+                <<"multi">>
```

Comment:
It seems like this could be a place to return an `enum` error instead of `required`, using the search query options macros from above to give the client an idea of the available query string parameters they could include.